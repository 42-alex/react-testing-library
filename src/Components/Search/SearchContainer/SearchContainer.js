import { useState } from 'react';
import Search from '../Search';


const SearchContainer = () => {

  const [search, setSearch] = useState('');

  const handleChange = (event) => {
    setSearch(event.target.value);
  }

  return (
    <div>
      <Search value={search} onChange={handleChange}>
        Search:
      </Search>

      <p>Looking for {search ? search : '...'}</p>
    </div>
  );

};

export default SearchContainer;
