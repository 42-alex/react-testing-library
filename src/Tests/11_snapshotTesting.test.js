import { render } from '@testing-library/react';
import SearchContainer from '../Components/Search/SearchContainer/SearchContainer';

describe('SearchContainer', () => {
  it('create snapshot of SearchContainer component', async () => {
    const { asFragment } = render(<SearchContainer />);

    expect(asFragment()).toMatchSnapshot();
  })
})
