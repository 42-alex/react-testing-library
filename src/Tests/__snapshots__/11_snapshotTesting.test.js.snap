// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer create snapshot of SearchContainer component 1`] = `
<DocumentFragment>
  <div>
    <div>
      <label
        for="search"
      >
        Search:
      </label>
      <input
        id="search"
        type="text"
        value=""
      />
    </div>
    <p>
      Looking for ...
    </p>
  </div>
</DocumentFragment>
`;
